app/src/main/java/org/dicekeys/app/encryption/AppKeystore.kt

```diff
@@ -79,29 +84,35 @@ class AppKeystore {
                 .setBlockModes(KeyProperties.BLOCK_MODE_CBC)
                 .setEncryptionPaddings(KeyProperties.ENCRYPTION_PADDING_PKCS7)
 
+        when(keystoreType){
+            KeystoreType.BIOMETRIC  -> {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                    builder.setUnlockedDeviceRequired(true)
+                }
 
-        if (isBiometric) {
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
-                builder.setUnlockedDeviceRequired(true)
-            }
-
-            builder.setUserAuthenticationRequired(true)
+                builder.setUserAuthenticationRequired(true)
 
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+                    builder.setUserAuthenticationParameters(0, KeyProperties.AUTH_BIOMETRIC_STRONG)
+                } else {
+                    builder.setUserAuthenticationValidityDurationSeconds(-1)
+                }
 
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
-                builder.setUserAuthenticationParameters(0, KeyProperties.AUTH_BIOMETRIC_STRONG)
-            } else {
-                builder.setUserAuthenticationValidityDurationSeconds(-1)
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                    builder.setInvalidatedByBiometricEnrollment(true)
+                }
             }
-
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-                builder.setInvalidatedByBiometricEnrollment(true)
+            KeystoreType.AUTHENTICATION -> {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                    builder.setUnlockedDeviceRequired(true)
+                }
             }
-        } else {
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
-                builder.setUserAuthenticationParameters(0, KeyProperties.AUTH_BIOMETRIC_STRONG)
-            } else {
-                builder.setUserAuthenticationValidityDurationSeconds(-1)
+            KeystoreType.KEYSTORE -> {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+                    builder.setUserAuthenticationParameters(0, KeyProperties.AUTH_BIOMETRIC_STRONG)
+                } else {
+                    builder.setUserAuthenticationValidityDurationSeconds(-1)
```

Comment:
What does this do?

Reply:
It forces to re-authenticate every time, it doesn't `cache` the authorization let's say.